mex/backend/merged/helpers.py

```diff
@@ -125,33 +141,49 @@ def search_merged_items_in_graph(
     Returns:
         MergedItemSearch instance
     """
-    # XXX We just search for extracted items and pretend they are already merged
-    #     as a stopgap for MX-1382.
     graph = GraphConnector.get()
-    result = graph.fetch_extracted_items(
+    result = graph.fetch_merged_items(
         query_string=query_string,
         stable_target_id=stable_target_id,
-        entity_type=(
-            None
-            if entity_type is None
-            else [t.replace("Merged", "Extracted") for t in entity_type]
-        ),
+        entity_type=entity_type,
         skip=skip,
         limit=limit,
     )
-    merged_model_adapter: TypeAdapter[AnyMergedModel] = TypeAdapter(
-        Annotated[AnyMergedModel, Field(discriminator="entityType")]
+    extracted_model_adapter: TypeAdapter[AnyExtractedModel] = TypeAdapter(
+        Annotated[AnyExtractedModel, Field(discriminator="entityType")]
     )
     items: list[AnyMergedModel] = []
     total: int = result["total"]
 
     for item in result["items"]:
-        item.pop("hadPrimarySource", None)
-        item.pop("identifierInPrimarySource", None)
-        item["identifier"] = item.pop("stableTargetId")
-        item["entityType"] = item["entityType"].replace("Extracted", "Merged")
+        extracted_items = [
+            extracted_model_adapter.validate_python(component)
+            for component in item["components"]
+            if component["entityType"] in EXTRACTED_MODEL_CLASSES_BY_NAME
+        ]
+
+        rules_raw = [
+            component
+            for component in item["components"]
+            if component["entityType"] in RULE_MODEL_CLASSES_BY_NAME
+        ]
+        if len(rules_raw) == 3:
+            rule_set_response = transform_raw_rules_to_rule_set_response(rules_raw)
+        elif len(rules_raw) == 0:
+            rule_set_response = None
+        else:
+            raise MExError(
```

Comment:
some test coverage here please

Reply:
done in d3208335baa4c8e4e9c38825c8e9941dbb06f464